backend/server/src/http.rs

```diff
@@ -43,21 +46,37 @@ pub(crate) async fn serve(
     // All our logic is encoded in the function `handle`. The only thing we are
     // doing here is to pass the context to that function, and clone its `Arc`
     // accordingly.
-    let factory = make_service_fn(move |_| {
-        let ctx = Arc::clone(&ctx);
-        async {
-            Ok::<_, Infallible>(service_fn(move |req| {
-                handle_panic(handle(req, Arc::clone(&ctx)))
-            }))
+    macro_rules! factory {
+        () => {
+            make_service_fn(move |_| {
+                let ctx = Arc::clone(&ctx);
+                async {
+                    Ok::<_, Infallible>(service_fn(move |req| {
+                        handle_panic(handle(req, Arc::clone(&ctx)))
+                    }))
+                }
+            })
         }
-    });
+    }
 
-    // Start the server with our service.
-    let addr = SocketAddr::new(config.address, config.port);
-    let server = Server::bind(&addr).serve(factory);
-    info!("Listening on http://{}", server.local_addr());
 
-    server.await?;
+    // Start the server with our service.
+    if let Some(unix_socket) = &config.unix_socket {
+        // Bind to Unix domain socket.
+        let path = Path::new(unix_socket);
+        if path.exists() {
+            fs::remove_file(path)?;
+        }
```

Comment:
Think about it as a socket, not a file. If the socket is open (the file exists), you cannot open a new one. Hence, the removal.

Reply:
Ohhh, interesting. And are we sure we want to remove the file instead of just erroring? If the TCP port is already taken, we just stop with an error. Shouldn't we do the same here? Or... the `remove_file` will probably fail if that socket is still used by another program, right? So the `remove_file` is just there for cleanup if the previous program didn't delete the socket file?